plugins/MantisCoreFormatting/core/MantisMarkdown.php

```diff
@@ -42,220 +42,212 @@
 class MantisMarkdown extends Parsedown
 {
 	/**
-	 * @var MantisMarkdown singleton instance for MantisMarkdown class.
+	 * singleton instance for MantisMarkdown class.
 	 */
-	private static $mantis_markdown = null;
+	private static ?MantisMarkdown $instance = null;
 
 	/**
-	 * @var string table class
+	 * CSS class for tables
 	 */
-	private $table_class = null;
+	private string $table_class = 'table table-nonfluid';
 
 	/**
-	 * @var string inline style
+	 * plugin configuration.
+	 * The value of the constant "ON" (1) or "OFF" (0).
 	 */
-	private $inline_style = null;
+	private int $config_process_buglinks = OFF;
 
 	/**
-	 * MantisMarkdown constructor.
+	 * plugin configuration
+	 * The value of the constant "ON" (1) or "OFF" (0).
 	 */
-	public function __construct() {
+	private int $config_process_urls = OFF;
 
-		# enable line break by default
-		$this->breaksEnabled = true;
+	/**
+	 * Collection of the captured code blocks.
+	 *
+	 * @var array<string, string>
+	 */
+	private array $codeblocks = [];
 
-		# set the table class
-		$this->table_class = 'table table-nonfluid';
+	public function __construct( ?int $p_process_urls = OFF, ?int $p_process_buglinks = OFF ) {
+		# Plugin configuration
+		if( in_array( $p_process_urls, [ OFF, ON ], true ) ) {
```

Comment:
The constructor is called from MantisCoreFormatting::formatted(), and parameters are coming from plugin_config_get() which does not check or enforce the return type.

While I'm generally in favor of strict type checking, considering that config options are basically user input, I'm concerned about the non-user-friendly behavior if we get non-integer data here (could actually be boolean, null, empty string...). This would trigger a TypeError exception, which is not user friendly.

Maybe cast plugin_config_get() return value to int when assigning $s_url / $s_buglinks.

Reply:
Is it user input? I think the values are already casted to an Integer at this point. The plugin handle the value and stores "1" or "0" with an type 1 into the database.

When `config_get` retrive the value it goes to this line

https://github.com/mantisbt/mantisbt/blob/aa93edbf9d45da731a24aa23875261d8cb78dfb7/core/config_api.php#L154C7-L156

```php
case CONFIG_TYPE_INT:
    return (int)$t_raw_value;
``` 

I want rely on this. If the values any other than the expected `OFF` or `ON`, something is broken and the exception is clear response to that.

Casting the values is some kind of silent failssafe that undermine the strictness. I understand your concerns, but think this situation is safe to be more strict.

Reply:
Or do i miss something?

Reply:
I reorganized the commits, not sure if I should open an issue for the commit "Add tests for plugin MantisCoreFormatting" to link it. I think I should, but very tired now.

No time today, so if you want the config values casted, please just do it. From my side i am fine to merge. Was a pleasure with you, thanks a lot.

Reply:
> Is it user input?

Well technically yes. It is submitted as  an HTML form, and furthermore can be edited manually from adm_config_report.php page, which makes it fairly easy to introduce an incorrect data type e.g. string (intentionally or not).

Of course if you only edit settings using the plugin's config page, then there can be no problem, and furthermore the config_edit.php page captures the submitted form using gpc_get_int() which ensures we are indeed getting a number. "Normal" usage of adm_config_report.php page also results in typecasted to int data.

So we're mostly safe unless the admin does something silly, in which case they should expect to get an error (and hopefully know how to analyze and fix it).

> If the values any other than the expected OFF or ON, something is broken and the exception is the clear response to that.

True, but my concern was that the error message is highly generic and misleading (INTERNAL APPLICATION ERROR) providing zero indication on the actual root cause unless you have $g_show_detailed_errors = ON, which is usually not the case in production systems.

Considering the low probability of this error occurring, let's leave it like that.